es-ds-components/components/es-zip-code-form.vue

```diff
@@ -0,0 +1,204 @@
+<script setup lang="ts">
+const props = defineProps({
+    constrained: {
+        type: Boolean,
+        default: false,
+    },
+    dark: {
+        type: Boolean,
+        default: false,
+    },
+    fieldName: {
+        type: String,
+        default: 'zip_code',
+    },
+    inputId: {
+        type: String,
+        required: true,
+    },
+    newTab: {
+        type: Boolean,
+        default: false,
+    },
+    placeholder: {
+        type: String,
+        default: 'ZIP code',
+    },
+    showPrivacySection: {
+        type: Boolean,
+        default: true,
+    },
+    privacyPolicyLink: {
+        type: String,
+        default: '',
+    },
+    privacyPolicyNewTab: {
+        type: Boolean,
+        default: false,
+    },
+    stackUntil: {
+        type: String,
+        default: '',
+    },
+    url: {
+        type: String,
+        required: true,
+    },
+    zipCodeValue: {
+        type: String,
+        default: '',
+    },
+    selectedProduct: {
+        type: String,
+        default: '',
+    },
+});
+
+const state = reactive({
+    zipCode: props.zipCodeValue,
+});
+
+watch(
```

Comment:
It looks like you're binding `state.zipCode` to the input, watching for that change, and updating the state accordingly.

I wonder if using:

```
const model = defineModel()
```

Then setting `v-mode=model` would be more optimal here

Reply:
You're right that setting this up as a `v-model` might have made sense for this too (although may get into messiness interacting with the validation setup above) - but I would rather not make that change, since it'd be an API change for downstreams and avoidable.  I don't remember exactly why it was originally set up this way ([PR for context](https://github.com/EnergySage/es-ds/pull/1385)), but it only needs one-way data binding as a prop rather than 2-way binding w upstream component.